### PR TITLE
redmine: 4.2.10 -> 5.0.5

### PR DESCRIPTION
**Pull request by megheaiulian**
###### Description of changes
Upgrade Redmine to 5.0.5

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by megheaiulian** [comment]
related to #231355

File: nixos/modules/services/misc/redmine.nix [hunk 1]
```diff
@@ -419,7 +419,7 @@ in
         Group = cfg.group;
         TimeoutSec = "300";
         WorkingDirectory = "${cfg.package}/share/redmine";
-        ExecStart="${bundle} exec rails server webrick -e production -p ${toString cfg.port} -P '${cfg.stateDir}/redmine.pid'";
+        ExecStart="${bundle} exec rails server -u webrick -e production -p ${toString cfg.port} -P '${cfg.stateDir}/redmine.pid'";
```
Comment (megheaiulian) [inline nixos/modules/services/misc/redmine.nix:422]: `rails` command has changed.

File: pkgs/applications/version-management/redmine/Gemfile [hunk 2]
```diff
@@ -34,54 +30,63 @@ gem 'tzinfo-data', platforms: [:mingw, :x64_mingw, :mswin]
...
 end
 
+# Optional CommonMark support, not for JRuby
+group :common_mark do
+  gem "commonmarker", '~> 0.23.8'
+  gem 'deckar01-task_list', '2.3.2'
+end
+
+# webrick
+gem 'webrick'
```
Comment (megheaiulian) [inline pkgs/applications/version-management/redmine/Gemfile:59]: `webrick` no longer included by default. had to add it.

File: pkgs/applications/version-management/redmine/Gemfile [hunk 3]
```diff
@@ -34,54 +30,63 @@ gem 'tzinfo-data', platforms: [:mingw, :x64_mingw, :mswin]
...
 end
 
 group :test do
   gem "rails-dom-testing"
-  gem 'mocha', (Gem.ruby_version < Gem::Version.new('2.7.0') ? ['>= 1.4.0', '< 2.0.0'] : '>= 1.4.0')
-  gem 'simplecov', '~> 0.18.5', :require => false
-  gem "ffi", platforms: [:mingw, :x64_mingw, :mswin]
+  gem 'mocha', '>= 1.4.0'
+  gem 'simplecov', '~> 0.22.0', :require => false
+  gem "ffi", platforms: [:mri, :mingw, :x64_mingw, :mswin]
```
Comment (megheaiulian) [inline pkgs/applications/version-management/redmine/Gemfile:76]: use native ffi.

**Comment by felixsinger** [comment]
Awesome! Thank you. I will have a look in a few hours.

File: pkgs/applications/version-management/redmine/default.nix [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
 let
-  version = "4.2.10";
+  version = "5.0.5";
   rubyEnv = bundlerEnv {
     name = "redmine-env-${version}";
 
     inherit ruby;
     gemdir = ./.;
-    groups = [ "development" "ldap" "markdown" "minimagick" "openid" "test" ];
+    groups = [ "development" "ldap" "markdown" "common_mark" "minimagick" "test" ];
```
Comment (felixsinger) [inline pkgs/applications/version-management/redmine/default.nix:10]: Just wondering, are the groups development and test needed?

**Review by felixsinger — APPROVED** [review]
Thank you! Looks good to me. Just one question.

**Comment by felixsinger** [comment]
@aanderse Could you have a look and maybe merge? It looks good to me. I also have tested it and everything works fine.

**Pull request merged**